### PR TITLE
Add new configuration options for OAuth2LoginSpec

### PR DESCRIPTION
**Pull request by nickbr23**
Improve Oauth2LoginSpec with:

`OAuth2LoginSpec.authorizationRequestResolver(ServerOAuth2AuthorizationRequestResolver authorizationRequestResolver)`
and
`OAuth2LoginSpec.authenticationMatcher(ServerWebExchangeMatcher authenticationMatcher)`

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@nickbr23 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=6462)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=6462) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
 import org.springframework.security.oauth2.client.web.server.OAuth2AuthorizationCodeGrantWebFilter;
```
Comment (jgrandja) [inline config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Please update Copyright header to 2019

Comment (nickbr23) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Done and updated the unit test in `OAuth2LoginTests`

Comment (jgrandja) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Awesome thanks! This looks great and it's ready to be merged. But before we merge can you please [squash](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#squash-commits) to 1 commit and update the commit message to follow this [format](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages).

Comment (nickbr23) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: All done.

**Comment by jgrandja** [comment]
Closed via b581bb7

**Comment by jgrandja** [comment]
Thank you @nickbr23! The commit message didn't exactly follow the expected format. I updated it for you. Please take a look so you know for the next time. I also added javadoc. Thanks again for taking this on and I look forward to seeing some more contributions from you :)

**Pull request closed without merging**